src/change_stream.ts

```diff
@@ -655,11 +629,22 @@ export class ChangeStream<
   hasNext(callback: Callback<boolean>): void;
   hasNext(callback?: Callback): Promise<boolean> | void {
     this._setIsIterator();
-    return maybePromise(callback, cb => {
-      this._getCursor((err, cursor) => {
-        if (err || !cursor) return cb(err); // failed to resume, raise an error
-        cursor.hasNext(cb);
-      });
+    return maybePromise(callback, callback => {
+      (async () => {
+        try {
+          const hasNext = await this.cursor.hasNext();
+          return callback(undefined, hasNext);
+        } catch (error) {
+          const errorOnResume = await this._processErrorAsync(error).catch(err => err);
```

Comment:
what is "errorOnResume" supposed to mean here? The name of `_processErrorAsync` doesn't really give a hint as to what it really does, but seems like what it would return is any error that is not resumable, so maybe `unresumableError` is a more apt name here (and in the other methods)?

Reply:
If there's any sort of error during the resume process, or if it's an unresumable error.  I'll see if I can find a better name that's more descriptive for both the method and the variable.